### DISCUSSION TITLE
Breadcrumb custom label & multidocs

### DISCUSSION THREAD

**Discussion by Dannyholt** [facebook/docusaurus#6786] [Ideas]
Is it possible to add something like this:

```markup
---
id: docusaurus-core
title: Docusaurus Client API
sidebar_label: Client API
breadcrumb_label: Docusaurus Client API
---
```

The reason for this is, we have multiple products and the opening or first doc for each product is named **Introduction** and so, this also appears in the breadcrumb. However, it doesn't really give you an indication which product you're viewing whereas a custom `breadcrumb_label` would.

For example:

```markup
---
id: product-a
title: Introduction to Product A
sidebar_label: Introduction
breadcrumb_label: Introduction to Product A
---
```

As for Multidocs, can we get an option to display the multidocs "home" in the breadcrumb.

At the moment, it displays like this:

```markup
Home (docs homepage) > Sidebar Item/Category > Doc
```

For multidocs, it should display like this imo.

```markup
Home (docs homepage) > Multidocs home directory/first article > Sidebar Item/Category > Doc
```

**Comment by slorber** [comment]
`breadcrumb_label` looks like a reasonable thing to add.

Note that the breadcrumb is highly related to the sidebar structure, so we probably also want to allow assigning a `breadcrumbLabel` attribute to some sidebar items too (docs & category). 

In the same way, there's already a `sidebar_label` frontmatter that maps to the sidebar `label` attribute of those items.

Does it make sense?

I'm assuming you want to use frontmatter over sidebars metadata because you are using an auto-generated sidebar?

--- 

About the "root" of the breadcrumb, for now, it's a quite simple solution linking to the site root (only if there's a root, some sites don't have one 😅 cf https://github.com/facebook/docusaurus/pull/6749)

We can definitively improve it too, however, it's not 100% clear what we should do by default and what options/API to expose to customize this behavior.

I like this by default:

```
Site Root > Docs Instance Root > Category1 > Category 2 > Doc
``` 

However:
- we currently don't have a way to provide a name/label for a docs plugin instance
- we currently don't force users to declare a docs root (slug: /), is taking the first doc of the first sidebar a good heuristic? 🤷‍♂️ 

As this breadcrumb feature is quite new, I'd to delay these possible improvements a bit and collect more user-feedback.

If you have opinions on how we should handle the breadcrumb root, please let us know below!

**Comment by Dannyholt** [comment]
@slorber 

We're not using auto-generated sidebars, but we don't want the breadcrumb label to be the same as the sidebar label, if that makes sense.

For multidocs, I like the idea of:

```markup
Site Root > Docs Instance Root > Category1 > Category 2 > Doc
```

As for providing a name/lable for the docs plugin instance. I am not entirely sure what would be the best solution here, but could it be something like this? 🤷‍♂️

```js
{
        id: 'community',
        path: 'community',
        routeBasePath: 'community',
        label: 'Community',
}
```

**Reply by slorber** [reply]
> We're not using auto-generated sidebars, but we don't want the breadcrumb label to be the same as the sidebar label, if that makes sense.

I mean having both `label` + `breadcrumbLabel` on sidebar items

---

One problem to also consider: sites with only one docs instance probably don't want to have the extra docs instance breadcrumb item. Not so easy to design such an API 🤪 

Also where should this label link to? "first doc of the first sidebar" may be a good default but it's not necessarily always what users will want? 🤷‍♂️ 

Maybe we could add a `breadcrumb: {to: "/myDoc",label: "My Doc"}` to plugin options and only add the item if it's present? 🤷‍♂️

**Reply by Dannyholt** [reply]
> Maybe we could add a breadcrumb: {to: "/myDoc",label: "My Doc"} to plugin options and only add the item if it's present? 🤷‍♂️

Yes, that could work! 👍

**Comment by ben-qnimble** [comment]
Expanding on slorber's  idea for a breadcrumb option to the doc's plugin, what if that option where an array that gets added to the start of a breadcrumb. Then users would have the option of breadcrumbs being:

Home -> Docs -> Cat1 -> Cat2 -> Doc

or 

Home -> Cat1 -> Cat2 -> Doc

or 

Docs -> Cat1 -> Cat2 -> Doc

by passing [Home, Docs] or [Home] or [Docs] in that array respectively? Additionally is this case the URL for Home and Docs could be set depending on the site's structure. (Don't need to assume Docs has a page with slug "/" etc)

**Reply by slorber** [reply]
yes that looks nice

So the default/fallback would be `breadcrumb: [{to: "/",label: "🏠"}]`? 
(only when there is a site homepage)

**Reply by slorber** [reply]
Let's continue this discussion in the dedicated issue: https://github.com/facebook/docusaurus/issues/6953

**Comment by slorber** [comment]
Let's continue this discussion in the dedicated issue: https://github.com/facebook/docusaurus/issues/6953
